lib/moonshot/parameter_collection.rb

```diff
@@ -17,6 +17,34 @@ def self.from_template(template)
       obj
     end
 
+    def self.apply_overrides(config)
+      raise unless config.instance_of?(Moonshot::ControllerConfig)
+
+      default_answer_file = File.join(config.project_root,
+                                      'moonshot',
+                                      'params',
+                                      "#{config.environment_name}.yml")
+
+      answer_file = config.answer_file ? config.answer_file : default_answer_file
+
+      # The order we override parameters in is, in order of least to most important:
+      #   * project_root/moonshot/params/#{environment}.yml
+      #   * file defined by the user with -a/--answer-file
+      #   * individual parameters the user input with -P/--parameter
```

Comment:
@pingram3030 Actually I was just thinking about this, does this mean that:

```
# 1. create stack with known param file containing 'FooParameter'
moonshot create -n staging

# 2. update FooParameter, -P takes precedence here
moonshot update -n staging -P FooParameter=newValue

# 3. update stack with new cloud formation template
moonshot update -n staging
```

I think that #3 would return the value of FooParameter to the value in the answer file the was changed by #2. If so, this breaks the pattern of allowing configuration change to live in AWS. We could change the algorithm to only use the by-name answer file for creation, and only use it for updates when a parameter is missing (such as a newly added parameter in an updated stack revision).